src/CoreManager.js

```diff
@@ -43,6 +43,24 @@ type ObjectController = {
   save: (object: ParseObject, options: RequestOptions) => ParsePromise;
   destroy: (object: ParseObject, options: RequestOptions) => ParsePromise;
 };
+type ObjectStateController = {
+  getState: (obj: ParseObject) => ?State;
+  initializeState: (obj: ParseObject, initial?: State) => State;
+  removeState: (obj: ParseObject) => ?State;
+  getServerData: (obj: ParseObject) => AttributeMap;
+  setServerData: (obj: ParseObject, attributes: AttributeMap) => void;
+  getPendingOps: (obj: ParseObject) => Array<OpsMap>;
+  setPendingOp: (obj: ParseObject, attr: string, op: ?Op) => void;
+  pushPendingState: (obj: ParseObject) => void;
+  popPendingState: (obj: ParseObject) => OpsMap;
+  mergeFirstPendingState: (obj: ParseObject) => void;
```

Comment:
Hmm, you sure this is correct? `mergeFirstPendingState` of `ObjectStateController` accepts `ParseObject`.....